tools/js-optimizer.js

```diff
@@ -7769,6 +7769,8 @@ function eliminateDeadGlobals(ast) {
 // it goes away.
 function JSDCE(ast) {
   var scopes = [{}]; // begin with empty toplevel scope
+  var isVarNameOrObjectKey;
```

Comment:
Thank you for reviewing.
I'll move the variable declaration close to the place used and add a comment what is done how.